### PR TITLE
Implement the resolver for helm chart

### PR DESCRIPTION
**Pull request by ywk253100**
Implement the resolver for helm chart

Signed-off-by: Wenkai Yin <yinw@vmware.com>

### CODE REVIEW

File: src/api/artifact/abstractor/resolver/chart/chart.go [hunk 1]
```diff
@@ -9,3 +13,68 @@
...
+	// ArtifactTypeChart defines the artifact type for helm chart
+	ArtifactTypeChart = "CHART"
+)
+
+func init() {
+	resolver := &resolver{
+		repoMgr:     repository.Mgr,
+		blobFetcher: blob.Fcher,
+	}
+	if err := resolv.Register(resolver, "application/vnd.cncf.helm.config.v1+json"); err != nil {
```
Comment (steven-zou) [inline src/api/artifact/abstractor/resolver/chart/chart.go]: For such value, we should have a set of const values for easy reference. Even including the value like `ArtifactTypeChart = "CHART"`

File: src/api/artifact/abstractor/resolver/chart/chart.go [hunk 2]
```diff
@@ -9,3 +13,68 @@
...
+	m := &v1.Manifest{}
+	if err := json.Unmarshal(manifest, m); err != nil {
+		return err
+	}
+	digest := m.Config.Digest.String()
+	layer, err := r.blobFetcher.FetchLayer(repository.Name, digest)
+	if err != nil {
+		return err
+	}
+	// TODO should we abstract all values?
```
Comment (steven-zou) [inline src/api/artifact/abstractor/resolver/chart/chart.go]: I think we should extract the data similar to the previous chart info block and put the all the left ones into the extra properties which are rendered as k-v pairs to the users

Comment (steven-zou) [reply src/api/artifact/abstractor/resolver/chart/chart.go]: Seems most of the info abstracted from the yaml can only be stored in the extra properties

File: src/api/artifact/abstractor/resolver/chart/chart.go (hunk 1 above)
Comment (ywk253100) [reply src/api/artifact/abstractor/resolver/chart/chart.go]: The media types of images are imported from distribution/oci packages, we didn't define them in Harbor.

For the `ArtifactTypeChart`, it's defined by the resolvers themselves, so define it in the resolver's package is reasonable

**Pull request merged**